test/plugins/iamauth.js

```diff
@@ -554,4 +557,51 @@ describe('#db IAMAuth Plugin', function() {
       assert.fail(`Unexpected reject: ${err}`);
     });
   });
+
+  it('successfully retries request on 500 IAM token service response and returns 200 response', function(done) {
+    if (process.env.NOCK_OFF) {
+      this.skip();
+    }
+
+    var iamMocks = nock(TOKEN_SERVER)
+      .post('/identity/token', {
+        'grant_type': 'urn:ibm:params:oauth:grant-type:apikey',
+        'response_type': 'cloud_iam',
+        'apikey': IAM_API_KEY
+      })
+      .times(4)
+      .reply(500, {error: 'internal_server_error', reason: 'Internal Server Error'})
+      .post('/identity/token', {
+        'grant_type': 'urn:ibm:params:oauth:grant-type:apikey',
+        'response_type': 'cloud_iam',
+        'apikey': IAM_API_KEY
+      })
+      .reply(200, MOCK_IAM_TOKEN_RESPONSE);
+
+    var cloudantMocks = nock(SERVER)
+      .post('/_iam_session', {access_token: MOCK_ACCESS_TOKEN})
+      .reply(200, {ok: true}, MOCK_SET_IAM_SESSION_HEADER)
+      .get(DBNAME)
+      .reply(200, {doc_count: 0});
+
+    var cloudantClient = new Client({ maxAttempt: 5, plugins: { iamauth: { iamApiKey: IAM_API_KEY } } });
```

Comment:
Any particular reason to test this for `5` and cause quite such a long test?
I mean we could do 1 or 2 or a shorter delay to make this a bit faster (and also potentially avoid increasing the test timeout).

Reply:
Nope, will reduce to `maxAttempt: 3`.